### PR TITLE
fix:- added elipsis and tooltip to the button content in table widget component

### PR DESCRIPTION
**Pull request by skjameela**
**Description**

[ Bug issue  ](https://github.com/appsmithorg/appsmith/issues/10278)

I have raised the pr inorder to adding a elipsis and tooltip for the button content in the table widget component to prevent the data hiding in overflow

**Screenshot** 

**Before issue resolved**

Screenshot from 2024-09-26 16-09-21

**After issue resolved** 

Screenshot from 2024-10-07 11-46-04

## Summary by CodeRabbit

- **New Features**
	- Enhanced table widget styling for improved responsiveness and interactivity.
	- Introduced tooltip functionality for buttons, displaying tooltips when text is truncated.
	- Added new styled components for better structure and visual appeal.

- **Bug Fixes**
	- Improved hover effects and conditional styling for table cells.

- **Tests**
	- Added unit tests for the `AutoToolTipComponent`, covering various tooltip scenarios and behavior.

### CODE REVIEW

**Comment by skjameela** [comment]
hi @rishabhrathod01 can you please review this pr

**Comment by rishabhrathod01** [comment]
@rahulbarwal could you please check this PR for review

**Comment by skjameela** [comment]
@rahulbarwal can you please review this

**Review by rahulbarwal — CHANGES_REQUESTED** [review]
There is a bug in flow, which makes the tooltip persist, resulting in 2 tooltips at same time. 
Screenshot 2024-10-17 at 4 32 08 PM

**Comment by skjameela** [comment]
@rahulbarwal  resolved the review comments can you please review now and that tooltip persistance i didnt written any code for that and for me it is working fine in local and checked with the deploy link also

**Comment by rahulbarwal** [comment]
@skjameela client lint is failing on latest changes, can you please fix them. 
https://github.com/appsmithorg/appsmith/actions/runs/11456348604/job/31874302881?pr=36925

**Comment by skjameela** [comment]
@rahulbarwal resoved linting issues  can you please check now

File: app/client/src/widgets/TableWidgetV2/component/cellComponents/AutoToolTipComponent.tsx [hunk 1]
```diff
@@ -36,45 +36,66 @@ const MAX_WIDTH = 500;
 const TOOLTIP_OPEN_DELAY = 500;
 const MAX_CHARS_ALLOWED_IN_TOOLTIP = 200;
 
-function useToolTip(children: React.ReactNode, title?: string) {
+function isButtonTextTruncated(element: HTMLElement) {
+  const spanElement = element.querySelector("span");
+  const offsetWidth = spanElement?.offsetWidth ?? 0;
+  const scrollWidth = spanElement?.scrollWidth ?? 0;
+
+  return scrollWidth > offsetWidth;
```
Comment (rahulbarwal) [inline app/client/src/widgets/TableWidgetV2/component/cellComponents/AutoToolTipComponent.tsx:44]: Check for null spanElement before accessing properties

If spanElement is null, comparison on line 44 will fail.

File: app/client/src/widgets/TableWidgetV2/component/cellComponents/AutoToolTipComponent.tsx [hunk 2]
```diff
@@ -160,11 +181,16 @@ function LinkWrapper(props: Props) {
 
 function AutoToolTipComponent(props: Props) {
   const content = useToolTip(props.children, props.title);
+  const buttonContent = useToolTip(props.children, props.title, true);
```
Comment (rahulbarwal) [inline app/client/src/widgets/TableWidgetV2/component/cellComponents/AutoToolTipComponent.tsx:184]: Isn't `content` variable on line 183 the same; why call `useTooltTip` twice?

`const content = useToolTip(props.children, props.title, props.columnType === ColumnTypes.BUTTON);`

This way, in one statement, you can get content for all the col types.

**Review by rahulbarwal — CHANGES_REQUESTED** [review]
Added some review comments.
Can we also add some tests(unit, preferably, to go along with these changes)?

**Comment by rahulbarwal** [comment]
@skjameela we have some linting and prettier failures. please check
https://github.com/appsmithorg/appsmith/actions/runs/11512801485/job/32048407888?pr=36925#step:9:5986

**Comment by skjameela** [comment]
@rahulbarwal is that linting issue because  for my pr? because i saw the error which contains many components i didnt work on those components

**Comment by rahulbarwal** [comment]
@skjameela I think you have made changes to this file: https://github.com/appsmithorg/appsmith/actions/runs/11517832347/job/32063355332?pr=36925#step:9:55

Also lets fix the typo in the name here: `AutoTooltipCpmponent`

To check, please run these commands locally to ensure issues are fixed
* yarn run prettier:ci
* yarn run lint:ci

**Comment by skjameela** [comment]
@rahulbarwal can you please check now i have resolved those linting and prettier issues

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**